### PR TITLE
Fix #311: 'CurrentFunction' and 'CurrentStruct' are causing errors in…

### LINKED ISSUE #311: `CurrentFunction` and `CurrentStruct` are causing errors in programs with more than 1 package
**Describe the bug**
The indexes used to refer to serialized struct and functions in packages do not match the indexes used in `CurrentFunction` and `CurrentStruct` fields of the package. For example, if there are 2 packages, one with 100 functions and another with 1, the `CurrentFunction` index of the second package is going to be 100, when it had to be 0 (the first and only one).

**To Reproduce**
1. Serialize and deserialize a CX program with a number of packages greater than 1, and each of these packages must have at least 1 function and/or struct definition.

**Expected behavior**
The indexes for the `CurrentFunction` and `CurrentStruct` of each serialized package must correspond to the indexes set in the deserialized program.

**Desktop (please complete the following information):**
 - CX Version [e.g. 0.6.2]

### PR DESCRIPTION
**Pull request by amherag**
… programs with more than 1 package.

Fixes #311 

Changes:
- `Deserialize` now resets the index of the CurrentPackage and CurrentFunction when a new package is entered.

Does this change need to mentioned in CHANGELOG.md?
Yes.

### CODE REVIEW

File: cx/serialize.go [hunk 1]
```diff
@@ -882,13 +884,16 @@ func dsPackages(s *sAll, prgrm *CXProgram) {
...
 		}
 
 		// CurrentStruct
 		if sPkg.StructsSize > 0 {
-			prgrm.Packages[i].CurrentStruct = prgrm.Packages[i].Structs[sPkg.CurrentStructOffset]
+			prgrm.Packages[i].CurrentStruct = prgrm.Packages[i].Structs[sPkg.CurrentStructOffset-strctCounter]
 		}
+
+		fnCounter += sPkg.FunctionsSize
+		strctCounter += sPkg.StructsSize
```
Comment (ingwal) [inline cx/serialize.go]: These two lines seem strange to me.  You increment a counter (counting pieces) with a size?  I would suggest that this is either a bug or a problem with the naming of one of the variables.

Comment (amherag) [reply cx/serialize.go]: `FunctionsSize` represents how many functions are in that package. The `fnCoqunter` is incremented by that number of functions to know how many functions in total we have deserialized so far.

Comment (ingwal) [reply cx/serialize.go]: That makes sense, but then the number of functions should not be called ...Size.  That is seriously misleading.

Can you make a quick change and then I'll approve and merge it.

Comment (amherag) [reply cx/serialize.go]: What name do you recommend? I'll change the other variable names for consistency.

Comment (ingwal) [reply cx/serialize.go]: Why not follow the pattern and call it FunctionCount and StructCount?

If I wrote this, I woudI call them NumFunctions and NumStructs, but that's not the established pattern right now.

**Pull request merged**